src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ConversationService.java

```diff
@@ -316,11 +316,11 @@ private void sendToConversationMembershipChannel(MetisCrudAction metisCrudAction
      * @return conversation if the user is a member
      */
     public Conversation mayInteractWithConversationElseThrow(Long conversationId, User user) {
-        Optional<Conversation> conversation = conversationRepository.findById(conversationId);
-        if (conversation.isEmpty() || !isMember(conversationId, user.getId())) {
+        Conversation conversation = conversationRepository.findWithConversationParticipantsByIdElseThrow(conversationId);
+        if (conversation.getConversationParticipants().stream().map(ConversationParticipant::getUser).noneMatch(user::equals)) {
```

Comment:
Did you consider avoiding this O(n) search and replace it by `.contains()`? Since ConversationParticipants are stored in a HashSet this can reduce runtime to O(1).